search_service/proxy/elasticsearch.py

```diff
@@ -98,7 +106,13 @@ def _get_search_result(self, page_index: int,
                 result = {}
                 for attr, val in es_payload.items():
                     if attr in model.get_attrs():
-                        result[attr] = val
+                        if attr in TAG_MAPPING:
+                            # maps a given badge or tag to a tag class
+                            result[attr] = [TAG_MAPPING[attr](tag_name=property_val)  # type: ignore
+                                            for property_val in val]
+                        else:
+                            result[attr] = val
+
```

Comment:
This seems like a good fit for factory method pattern -- a method generates object given attribute -- if it decouples from Tag class for it's currently expecting `tag_name` Can we abstract this out?

Reply:
done!